### PR TITLE
electrostatic simplify

### PR DESCRIPTION
**Pull request by ssskkkky**
downgrade to electrostatic when beta_e=0

### CODE REVIEW

File: include/solver.h [hunk 1]
```diff
@@ -185,11 +187,20 @@ class EigenSolver {
 
     void matrixAssembler(matrix_type& mat) {
 #ifdef EMME_DEBUG
-        if (mat.getRows() != 2 * grid_info.npoints ||
-            mat.getCols() != 2 * grid_info.npoints) {
-            throw std::runtime_error(
-                "Matrix dimension and grid length mismatch.");
+        if (std::fpclassify(para.beta_e) == FP_ZERO) {
```
Comment (12ff54e) [inline include/solver.h]: This if is redundant, just use dim here

File: include/solver.h [hunk 2]
```diff
@@ -204,48 +215,76 @@ class EigenSolver {
...
-            for (unsigned int j = i; j < grid_info.npoints; j++) {
-                if (i == j) {
-                    mat(i, j) = (1.0 + 1.0 / para.tau);
-                    mat(i, j + grid_info.npoints) = 0.0;
-                    mat(i + grid_info.npoints, j) = 0.0;
-                    mat(i + grid_info.npoints, j + grid_info.npoints) =
-                        (2.0 * para.tau) / para.beta_e *
-                        para.bi(grid_info.grid[i]);
-                } else {
+        if (std::fpclassify(para.beta_e) == FP_ZERO) {
```
Comment (12ff54e) [inline include/solver.h:210]: This if is better put into the inner for loop to make code cleaner.

**Review by 12ff54e — CHANGES_REQUESTED** [review]
Check the inline comments pls

File: include/solver.h (hunk 2 above)
Comment (ssskkkky) [reply include/solver.h:210]: In the loop we need to repeat the if block for twice in both diagonal and Non-diagonal entries.

Comment (12ff54e) [reply include/solver.h:210]: fine

**Review by 12ff54e — APPROVED** [review]

**Pull request merged**